pkg/client_data/lib/account_api.dart

```diff
@@ -2,11 +2,51 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:convert';
+
 import 'package:json_annotation/json_annotation.dart';
 import 'package:meta/meta.dart';
 
 part 'account_api.g.dart';
 
+/// Session data to be cached from the authenticated client.
+@JsonSerializable()
+class ClientSessionData {
+  final String imageUrl;
+
+  ClientSessionData({
+    @required this.imageUrl,
+  });
+
+  factory ClientSessionData.fromJson(Map<String, dynamic> json) =>
+      _$ClientSessionDataFromJson(json);
+
+  Map<String, dynamic> toJson() => _$ClientSessionDataToJson(this);
+}
+
+/// The server-provided status of the current session.
+@JsonSerializable()
+class ClientSessionStatus {
+  final bool changed;
+  final DateTime expires;
+
+  ClientSessionStatus({
+    @required this.changed,
+    @required this.expires,
```

Comment:
Can we just expose `sessionChanged: bool` and `sessionIsValid: bool` to keep to reduce information leak...

(Feel free to suggest better names)

Reply:
I don't mind `sessionChanged` (vs the current `changed`), although as the object is `*SessionStatus`, it feels like we are better off with the shorter name.

`[session]IsValid` is a good question: there is not much value keeping it there, because I don't see a scenario where the `changed` doesn't trigger a page reload and the flag is useful (== we can meaningfully act on it, because it provides more information than we have already known). On the other hand, `expires` does have a useful scenario: we can trigger a popup before the session expires, preventing e.g. sessions where people would otherwise lose their status that they were about to send to the server.

Reply:
> On the other hand, expires does have a useful scenario: we can trigger a popup before the session expires, preventing e.g. sessions where people would otherwise lose their status that they were about to send to the server.

Good point, it means we can re-create the session if the client-side is still signed in...